hscontrol/db/ip.go

```diff
@@ -337,3 +337,12 @@ func (db *HSDatabase) BackfillNodeIPs(i *IPAllocator) ([]string, error) {
 
 	return ret, err
 }
+
+func (i *IPAllocator) RemoveUsedIps(ips []netip.Addr) {
```

Comment:
I had to read this a few times for it to make sense to me. 

We are not really "removing" it, we are kind of adding it back, or freeing it.

Maybe:

```suggestion
func (i *IPAllocator) FreeIPs(ips []netip.Addr) {
```

Or very technical, maybe `DeallocateIPs`